src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -0,0 +1,89 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands.autonomousCommands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.HopperController;
+import frc.robot.subsystems.IndexerController;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.shooter.ShooterSpeedController;
+import frc.robot.subsystems.shooter.ShooterMath;
+
+/**
+ * Add your docs here.
+ */
+public class AutonomousShooterCommand extends CommandBase{
+
+    private Drive drive;
+    private ShooterSpeedController shooterSpeedController;
+    private HopperController hopperController;
+    private IndexerController indexerController;
+    //variables
+    private double ballsShot = 0;
+    private double distanceToGoal = 149.0; 
+    private boolean hasBall = false;
+    //constants
+    private final double INNER_DISTANCE_FROM_TARGET = 29.0;
+
+    public AutonomousShooterCommand(Drive drive, ShooterSpeedController shooterSpeedController,
+    HopperController hopperController, IndexerController indexerController){
+        this.drive = drive;
+        this.shooterSpeedController = shooterSpeedController;
+        this.hopperController = hopperController;
+        this.indexerController = indexerController;
+
+        addRequirements(drive);
+    }
+
+    @Override
+    public void initialize(){
+        System.out.println("Shooter initialized");
+    }
+
+    @Override
+    public void execute(){
+        shooterSpeedController.setLaunchSpeed(ShooterMath.determineLaunchSpeed(distanceToGoal + INNER_DISTANCE_FROM_TARGET));
+
+        for(int ballsShot = 0 ; ballsShot < 3 ; ballsShot ++){
```

Comment:
I don't think you need a for loop here at all.  You can get rid of it, but keep what's inside of it.

Also, since you declare a local ballsShot right here in the for loop (let's call that **ballsShot**, that's what gets incremented, not the ballsShot (which we'll call _ballsShot_) that you declare at the top of the class.  You're isFinished check will never return true because the class _ballsShot_ isn't incremented in the for loop, only **ballsShot** is.

Reply:
okay, I'll get rid of the for loop